### PR TITLE
build: update build_depends.repos

### PR DESCRIPTION
**Pull request by kenji-miyake**
## Description

Update build_depends.repos for Humble.
Also, some refactorings following https://github.com/autowarefoundation/autoware/pull/374 were applied.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by kenji-miyake** [comment]
We have to wait for nav2 to be released probably today.

File: build_depends.humble.repos [hunk 1]
```diff
@@ -1,35 +1,31 @@
...
-  universe/tier4_autoware_msgs:
+  universe/external/tier4_autoware_msgs:
     type: git
-    url: https://github.com/tier4/AutowareArchitectureProposal_msgs.git # TODO(Tier IV): Rename to tier4/tier4_autoware_msgs
+    url: https://github.com/tier4/tier4_autoware_msgs.git
     version: tier4/universe
-  universe/vendor/grid_map:
+  universe/external/grid_map: # TODO: Use upstream after https://github.com/ANYbotics/grid_map/pull/349 is merged
     type: git
     url: https://github.com/tier4/grid_map.git
```
Comment (wep21) [inline build_depends.humble.repos:22]: Now, you can use the upstream repository because https://github.com/ANYbotics/grid_map/pull/349 has been merged.

Comment (kenji-miyake) [reply build_depends.humble.repos:22]: Ah, exactly!

Comment (kenji-miyake) [reply build_depends.humble.repos]: 89765a1f9143ef1ccf062f7d408d94884596d28e
d7efaf6c593a4edaca83fc1091911ebbd2bc067f

**Comment by kenji-miyake** [comment]
@wep21 @h-ohta Could you approve this if there is no issue left? :pray:

**Review by wep21 — APPROVED** [review]
LGTM

**Pull request merged**